main.cpp

```diff
@@ -225,6 +235,20 @@ IntType& IntType::operator*=(const int other)
     return *this;
 }
 
+IntType& IntType::apply(std::function<IntType&(int&)> intFunc)
+{
+    if(intFunc)
+    intFunc(*intTypeHeap);
+    return *this;
+}
+
+IntType& IntType::apply(void(*intFunc) (int&))
+{
+    if(intFunc)
+    intFunc(*intTypeHeap);
+    return *this;
+}
```

Comment:
- [ ] Fix the Formatting
Also, what does your `std::function` return?

Reply:
it returns a reference to intType, to allow chaining